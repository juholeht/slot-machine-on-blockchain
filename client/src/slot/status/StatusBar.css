.account {
    text-align: left;
    padding-left: 3.7rem;
    color: white;
    position: absolute;
    z-index: 1;
    width: 100%;
    padding-top: 1.7rem;
  }

.balances-row {
    display: flex;
    padding-top: 0.4rem;
}

.address {
    padding-top: 1rem;
}

.winnings {
    padding-left: 8rem;
}

.jackpot {
    padding-left: 8rem;
}