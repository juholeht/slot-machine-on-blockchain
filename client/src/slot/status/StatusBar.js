import {
  React,
} from 'react';
import { defineMessage, useIntl, injectIntl } from 'react-intl';

import PropTypes from 'prop-types';
import { ETH_CURRENCY_CHAR } from '../constant';

import './StatusBar.css';

const STATUS_BAR_MESSAGES = {
  addressText: defineMessage({
    id: 'addressText',
    description: 'Text for player address',
    defaultMessage: 'Address',
  }),
  balanceText: defineMessage({
    id: 'balanceText',
    description: 'Text for player balance',
    defaultMessage: 'Balance',
  }),
  winningsText: defineMessage({
    id: 'winningsText',
    description: 'Text for player winnings',
    defaultMessage: 'Winnings',
  }),
  jackpotText: defineMessage({
    id: 'jackpotText',
    description: 'Text for Jackpot prize pool',
    defaultMessage: 'Jackpot',
  }),
};

export const StatusBar = ({
  address, balance, winnings, jackpotPrize,
}) => {
  const intl = useIntl();

  return (
    <div className="account">
      <div className="address">
        {`${intl.formatMessage(STATUS_BAR_MESSAGES.addressText)}: ${address}`}
      </div>
      <div className="balances-row">
        <div className="balance">
          {`${intl.formatMessage(STATUS_BAR_MESSAGES.balanceText)}: ${balance} ${ETH_CURRENCY_CHAR}`}
        </div>
        <div className="winnings">
          {`${intl.formatMessage(STATUS_BAR_MESSAGES.winningsText)}: ${winnings} ${ETH_CURRENCY_CHAR}`}
        </div>
        <div className="jackpot">
          {`${intl.formatMessage(STATUS_BAR_MESSAGES.jackpotText)}: ${jackpotPrize} ${ETH_CURRENCY_CHAR}`}
        </div>
      </div>
    </div>
  );
};

StatusBar.propTypes = {
  address: PropTypes.string.isRequired,
  balance: PropTypes.number.isRequired,
  winnings: PropTypes.string.isRequired,
  jackpotPrize: PropTypes.string.isRequired,
};

export default injectIntl(StatusBar);
