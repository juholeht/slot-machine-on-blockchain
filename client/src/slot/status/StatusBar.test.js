import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { StatusBar } from './StatusBar';

configure({ adapter: new Adapter() });

jest.mock('react-intl', () => {
  const reactIntl = jest.requireActual('react-intl');
  const intl = reactIntl.createIntl({
    locale: 'en',
  });
  return {
    ...reactIntl,
    useIntl: () => intl,
  };
});

describe('<StatusBar />', () => {
  it('renders properly with default values', () => {
    const wrapper = shallow(<StatusBar
      address="0xfoobar"
      balance={0.62}
      winnings={0.0044}
      jackpotPrize={12.4234}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
