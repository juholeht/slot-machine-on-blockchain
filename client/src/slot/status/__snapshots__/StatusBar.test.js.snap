// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatusBar /> renders properly with default values 1`] = `
<div
  className="account"
>
  <div
    className="address"
  >
    Address: 0xfoobar
  </div>
  <div
    className="balances-row"
  >
    <div
      className="balance"
    >
      Balance: 0.62 Ξ
    </div>
    <div
      className="winnings"
    >
      Winnings: 0.0044 Ξ
    </div>
    <div
      className="jackpot"
    >
      Jackpot: 12.4234 Ξ
    </div>
  </div>
</div>
`;
